registry/storage/driver/s3-aws/s3.go

```diff
@@ -1010,6 +1011,12 @@ ListLoop:
 		if err != nil {
 			return err
 		}
+		if output.Errors != nil && len(output.Errors) > 0 {
+			// ideally all errors would be returned in some way
+			// until then, at least pass back the first error code
+			oErr := output.Errors[0]
```

Comment:
Have you looked closer into the errors produced here to if we could do something like extract the object paths to aggregate into a custom error type that includes all failed paths? Based on how errors in the results get serialized,
```
 <Error>
  <Key>sample2.txt</Key>
  <Code>AccessDenied</Code>
  <Message>Access Denied</Message>
 </Error>
```
this seems like it should be possible.